elixir/apps/web/lib/web/components/form_components.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Web.FormComponents do
+  @moduledoc """
+  Provides components that can be shared across forms.
+  """
+  use Phoenix.Component
+  use Web, :verified_routes
+  import Web.CoreComponents
+
+  @doc """
+  Conditionally renders form fields corresponding to a given provisioning strategy type.
+
+  ## Examples
+
+    <.provisioning_strategy_form form={%{
+      provisioning_strategy: "jit",
+      jit_user_filter_type: "email_allowlist",
+      jit_user_filter_value: "jamil@foo.dev,andrew@foo.dev",
+      jit_extract_groups: true
+    }} />
+
+    <.provisioning_strategy_form form={@form} />
+  """
+  attr :form, :map, required: true, doc: "The form to which this component belongs."
+
+  def provisioning_strategy_form(assigns) do
+    ~H"""
+    <h2 class="mb-4 text-xl font-bold text-gray-900 dark:text-white">Provisioning strategy</h2>
+    <ul class="mb-4 w-full sm:flex border border-gray-200 rounded-lg dark:bg-gray-700 dark:border-gray-600 dark:text-white">
+      <li class="w-full border-b border-gray-200 sm:border-b-0 sm:border-r dark:border-gray-600">
+        <div class="text-lg font-medium p-3">
+          <.input
+            id="provisioning_strategy_jit"
+            label="Just-in-time"
+            type="radio"
+            value="jit"
+            field={@form[:provisioning_strategy]}
+            checked={@form[:provisioning_strategy].value == "jit"}
+            required
+          />
+        </div>
+        <p class="px-4 py-2 text-sm text-gray-500 dark:text-gray-400">
+          Provision users and groups on the fly when they first sign in.
+        </p>
+      </li>
+      <li class="w-full border-b border-gray-200 sm:border-b-0 sm:border-r dark:border-gray-600">
+        <div class="text-lg font-medium p-3">
+          <.input
+            id="provisioning_strategy_scim"
+            label="SCIM 2.0"
+            type="radio"
+            value="scim"
+            field={@form[:provisioning_strategy]}
+            checked={@form[:provisioning_strategy].value == "scim"}
+            required
+          />
+        </div>
+        <p class="px-4 py-2 text-sm text-gray-500 dark:text-gray-400">
+          Provision users using the SCIM 2.0 protocol. Requires a supported identity provider.
+        </p>
+      </li>
+      <li class="w-full border-b border-gray-200 sm:border-b-0 sm:border-r dark:border-gray-600">
+        <div class="text-lg font-medium p-3">
+          <.input
+            id="provisioning_strategy_manual"
+            label="Manual"
+            type="radio"
+            value="manual"
+            field={@form[:provisioning_strategy]}
+            checked={@form[:provisioning_strategy].value == "manual"}
+            required
+          />
+        </div>
+        <p class="px-4 py-2 text-sm text-gray-500 dark:text-gray-400">
+          Disable automatic provisioning and manually manage users and groups.
+        </p>
+      </li>
+    </ul>
+    <%= if @form[:provisioning_strategy].value == "jit" do %>
```

Comment:
@AndrewDryga

Does this UI accomplish what you're planning for the backend? See screenshots in the issue description for more detail. I decided to decouple the provisioning strategy from the authentication method since some providers support SCIM with both OIDC and SAML (Azure, Authentik).

Reply:
@jamilbk I did not plan to implement SCIM for OIDC (since every provider that supports that also supports SAML), but the general structure looks good.

Reply:
@AndrewDryga Gotcha -- it could be possible that our SCIM implementation doesn't require much, if any work to be auth method agnostic since it's just a REST API essentially. I think if the resource IDs match (e.g. `sub` in OIDC and `user.id` in SAML), it should theoretically work for both types of providers.

Reply:
If it turns out to come to us easy, it might be a good idea to be auth method agnostic because of that pesky admin that wants SCIM with their OIDC connector in Azure.

Reply:
The more trickier part would be implementing JIT provisioning for SAML (parsing XML doesn't seem fun), but maybe we can just disable that for now.

Reply:
Followed up over DM. SCIM is going to be tedious. Will approach it in stages and disable SCIM option until ready.